src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1189,19 +1194,46 @@ public Component getTableCellRendererComponent(JTable table, Object value,
                 Icon icon = chooser.getIcon(file);
                 setIcon(icon);
 
-            } else if (value instanceof Long) {
-                long len = ((Long) value) / 1024L;
+            } else if (value instanceof Long len) {
                 if (listViewWindowsStyle) {
-                    text = MessageFormat.format(kiloByteString, len + 1);
-                } else if (len < 1024L) {
-                    text = MessageFormat.format(kiloByteString, (len == 0L) ? 1L : len);
+                    updateMessageFormatPattern(kiloByteString, 1);
```

Comment:
Now you always call `updateMessageFormatPattern` with 1 as the second parameter. Maybe you could initialise both MessageFormat and NumberFormat once and then just use the objects. Or switch back to `MessageFormat.format` method.

Reply:
MessageFormat.format method was truncating "0" from values like "0.0", "1.0" and so on.
Due to that I initialize MessageFormat and NumberFormat instance.

The first parameter in updateMessageFormatPattern  method changes based on file size, so calling it multiple times to update the pattern.
The second parameter can be removed.

Reply:
> MessageFormat.format method was truncating "0" from values like "0.0", "1.0" and so on. Due to that I initialize MessageFormat and NumberFormat instance.

Got it.

> The first parameter in updateMessageFormatPattern method changes based on file size, so calling it multiple times to update the pattern. The second parameter can be removed.

Right, it makes sense. Please remove the second parameter.

After the second parameter is removed, `NumberFormat` does not change though, so it can be initialised once.

Reply:
Yeah sure, I will remove the second parameter.